File: Programming101/week0/is_increasing.py
def is_increasing(seq):
    increasing = True
    for i in range(len(seq) - 1):
        if seq[i] >= seq[i+1]:
            increasing = False
    return increasing


def main():
    print(is_increasing([1, 2, 3, 4, 5]))
    print(is_increasing([1]))
    print(is_increasing([5, 6, -10]))
    print(is_increasing([1, 1, 1, 1]))


if __name__ == '__main__':
    main()

File: FrontendJavaScript/week0/2-A-Set-Of-Problems/set1/isIncreasing.js
"use strict";

var isIncreasing = function(seq) {
  return seq.every(function(element, index) {
    if (index+1 === seq.length) {
      return true;
    }
    return seq[index] < seq[index+1];
  });
};

console.log(isIncreasing([1,2,3,4,5]) === true);
console.log(isIncreasing([1]) === true);
console.log(isIncreasing([5,6,-10]) === false);
console.log(isIncreasing([1,1,1,1]) === false);
